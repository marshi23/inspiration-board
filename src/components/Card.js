import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';


// class Card extends Component {
const Card = (props) => {

    return (
      <div className="card">
        <section className="card__content">
          <p className="card__content-text">{props.text}</p>

          {props.emoji && (
              <p className="card__content-emoji">
              {emoji.getUnicode(props.emoji)}
              </p>)}

          <button onClick={() => props.deleteCardCallback(props.id)} type="button" className="card__delete">X</button>
        </section>
      </div>
    )
  }


Card.propTypes = {
  id: PropTypes.number,
  text: PropTypes.string,
  emoji: PropTypes.string,
  deleteCardCallback: PropTypes.func,
};

export default Card;
