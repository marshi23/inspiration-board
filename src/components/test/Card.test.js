import React from 'react';
import Card from '../Card';
import { shallow } from 'enzyme';


describe('Card', () => {
  it ('test it matches and exisiting snapshot', () => {
    const wrapper = shallow(
      <Card
        id={1}
        text="hello there"
        emoji="heart_eyes"
        deleteCardCallback={() => {}}
      />);

      expect(wrapper).toMatchSnapshot();
  });

});
