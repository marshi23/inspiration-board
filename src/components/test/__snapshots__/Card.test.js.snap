// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card test it matches and exisiting snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    deleteCardCallback={[Function]}
    emoji="heart_eyes"
    id={1}
    text="hello there"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="card__content"
      >
        <p
          className="card__content-text"
        >
          hello there
        </p>
        <p
          className="card__content-emoji"
        >
          üòç
        </p>
        <button
          className="card__delete"
          onClick={[Function]}
          type="button"
        >
          X
        </button>
      </section>,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="card__content-text"
          >
            hello there
          </p>,
          <p
            className="card__content-emoji"
          >
            üòç
          </p>,
          <button
            className="card__delete"
            onClick={[Function]}
            type="button"
          >
            X
          </button>,
        ],
        "className": "card__content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "hello there",
            "className": "card__content-text",
          },
          "ref": null,
          "rendered": "hello there",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "üòç",
            "className": "card__content-emoji",
          },
          "ref": null,
          "rendered": "üòç",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "X",
            "className": "card__delete",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "X",
          "type": "button",
        },
      ],
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="card__content"
        >
          <p
            className="card__content-text"
          >
            hello there
          </p>
          <p
            className="card__content-emoji"
          >
            üòç
          </p>
          <button
            className="card__delete"
            onClick={[Function]}
            type="button"
          >
            X
          </button>
        </section>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="card__content-text"
            >
              hello there
            </p>,
            <p
              className="card__content-emoji"
            >
              üòç
            </p>,
            <button
              className="card__delete"
              onClick={[Function]}
              type="button"
            >
              X
            </button>,
          ],
          "className": "card__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "hello there",
              "className": "card__content-text",
            },
            "ref": null,
            "rendered": "hello there",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "üòç",
              "className": "card__content-emoji",
            },
            "ref": null,
            "rendered": "üòç",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "X",
              "className": "card__delete",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "X",
            "type": "button",
          },
        ],
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
